### DISCUSSION TITLE
Simulation of a 50 year old calculator implemented as a state machine

### DISCUSSION THREAD

**Discussion by jongena** [hneemann/Digital#1140] [Q&A]
If anyone is interested, I finalized the simulation of a 1970s Japanese calculator, implemented as a state machine. The simulation runs up to a clock frequency of 45 kHz. possibly also higher. This mimicks very closely the calculation speed of the real calculator. Very nice to see the full internal logical clockwork in operation. 
The necessary Digital files can be sent on request. 

J.Ongena

**Comment by mengstr** [comment]
Sounds interesting. I'd love to have a look at it - and maybe also wire it up in real hardware.  I do have about 20 000 of 1980's TTL ICs and several hundred EPROMS in various sizes that wants to be used for something :)

**Reply by jongena** [reply]
The hardware version exists, it is a Riccar calculator made by TEAL in Tokyo in 1971. It uses now defunct Japanese Logic ICs working with 0 and -24V.... The logic simulation results from reverse engineering of the broken real example. The calculator is repaired and functional again. The full story and schematic can be found at http://madrona.ca/e/eec/calcs/TEALRiccar.html. The theory of operation can be found under the Royal Digital Calculator http://madrona.ca/e/eec/calcs/TEALRoyalDI.html, which is another type of this series of calculators made by TEAL. 
If you can pass me your email adres I will send the Digital files. It took me a while to get the simulation going, until I understood the reason of existence of the Reset component in Digital. Using this component in the right way in the various flip-flops (for more info, see the discussion "Circuit with random start and how to avoid ? #1063" - with final solution at the very end of that discussion) eliminated the randomness when starting up the simulation (and oscillations during the first versions of the simulation). Any comment on how to further improve the current version of the simulation would always be welcome.

**Reply by mengstr** [reply]
Nice, thank you. Please send them to **mats.engstrom** at the domain **protonmail.ch**

**Reply by jongena** [reply]
Dear Mats, I have sent you the files by email just now !

**Reply by mengstr** [reply]
Thank you Józef. I've received the files and started up the sim just to verify that everything is there...

**Comment by fredcwbr** [comment]
I'm using Digital to create some educational purpose schetches, including a double access RAM arbiter (work in progress) ; If you could share your files, I would appreciate it very much, and if there is no problem would also request yout authorization for its use as   an educational tool , with all the references and credits granted to thee.
Thanks in advance   -  fredcw at gmail dot com

Best regards from Brazil,

And congrats for your fine work.
